Core/GDCore/IDE/Events/EventsRefactorer.cpp

```diff
@@ -677,6 +686,36 @@ bool EventsRefactorer::ReplaceStringInConditions(
   return somethingModified;
 }
 
+bool EventsRefactorer::ReplaceStringInStringEvent(gd::ObjectsContainer& project,
+                                                  gd::ObjectsContainer& layout,
+                                                  gd::BaseEvent& event,
+                                                  gd::String toReplace,
+                                                  gd::String newString,
+                                                  bool matchCase) {
+  bool somethingModified = false;
+  vector<gd::String> stringEvent = event.GetAllSearchableStrings();
+
+  for (std::size_t sNb = 0; sNb < stringEvent.size(); ++sNb) {
+    gd::String newStringEvent =
+        matchCase ? stringEvent[sNb].FindAndReplace(toReplace, newString, true)
+                  : ReplaceAllOccurencesCaseUnsensitive(
+                        stringEvent[sNb], toReplace, newString);
+
+    if (newStringEvent != stringEvent[sNb]) {
+      if (event.GetType() == "BuiltinCommonInstructions::Group") {
+        gd::GroupEvent& groupEvent = dynamic_cast<gd::GroupEvent&>(event);
+        groupEvent.SetName(newStringEvent);
+      } else if (event.GetType() == "BuiltinCommonInstructions::Comment") {
+        gd::CommentEvent& commentEvent = dynamic_cast<gd::CommentEvent&>(event);
+        commentEvent.SetComment(newStringEvent);
+      }

```

Comment:
This seems a good place where we can use "polymorphism" instead of a switch case (https://refactoring.guru/replace-conditional-with-polymorphism).

In other words: 
because you can't really know from the outside of many strings there are and how to replace them, you can create a method doing this whole job inside BaseEvent (by default, it does nothing) and then redefine it for each event.

You can call this method: `ReplaceInAllSearchableStrings` or something similar. It should I guess takes the old string, new string, and whether to match case (and return true if something was modified).

The advantage: in the future, you only need to redefine this method if we create a new event type and it will work automatically.
The disadvantage: you put something related to "refactoring" events inside the event itself... but well that's just a search/replace, it makes sense to have this in an event class :)